components/steps/Steps.tsx

```diff
@@ -1,17 +1,23 @@
-import Step1 from '@/components/steps/Step1'
-import Step2 from '@/components/steps/Step2'
-import Step3 from '@/components/steps/Step3'
-import Step4 from '@/components/steps/Step4'
+import { Step1Icon, Step2Icon, Step3Icon, Step4Icon, useStepIcon } from '@/components/steps/icons'
 
-export default function Steps() {
+type StepsProps = {
+  activeStep: number
+}
+
+export function Steps({ activeStep }: StepsProps) {
```

Comment:
We can accept the `activeStep` number from the outer component so that this outer component tracks the value, and the Step component just displays what is has been asked to.